apps/gpc-verifier/app/util/verifyProof.ts

```diff
@@ -0,0 +1,107 @@
+import { Dispatch } from "react";
+import {
+  GPCBoundConfig,
+  GPCProof,
+  gpcArtifactDownloadURL,
+  gpcVerify,
+  deserializeGPCBoundConfig,
+  deserializeGPCRevealedClaims,
+  PODMembershipLists
+} from "@pcd/gpc";
+import { tryRecordNullifierHash } from "@/util/persistence";
+
+export const verifyProof = async (
+  boundConfig: GPCBoundConfig,
+  membershipLists: PODMembershipLists,
+  externalNullifier: string,
+  watermark: string,
+  proofStr: string,
+  setVerified: Dispatch<boolean>
+) => {
+  try {
+    if (!proofStr) {
+      throw new Error("Proof cannot be empty!");
+    }
+
+    const proofObj = JSON.parse(proofStr);
+    const vProof = JSON.parse(proofObj.proof) as GPCProof;
+
+    // Use the boundConfig we provided.
+    // However, gpcBindConfig might not always pick the same circuit as gpcProve,
+    // since it doesn't know the size of the inputs.
+    // Here we would like to use the circuitIdentifier returned by gpcProve.
+    const vConfig = {
+      ...boundConfig,
+      circuitIdentifier: deserializeGPCBoundConfig(proofObj.config)
+        .circuitIdentifier
+    };
+
+    // Make sure the membershipLists in the revealed claims matches
```

Comment:
Similar to the above, this is a good trick, but needs a bit more explanation.  We're not literally making sure they match, but forcing the verifier to use our preferred list, with the understanding that it will force the verification to fail if it wasn't the same as the one used to generate the proof.